lib/core/util.js

```diff
@@ -26,6 +26,57 @@ function isBlobLike (object) {
   )
 }
 
+function encode (val) {
+  return encodeURIComponent(val).replace(/%3A/gi, ':').replace(/%24/g, '$').replace(/%2C/gi, ',').replace(/%20/g, '+').replace(/%5B/gi, '[').replace(/%5D/gi, ']')
+}
+
+function isDate (val) {
+  return toString.call(val) === '[object Date]'
+}
+
+function isObject (val) {
+  return val !== null && typeof val === 'object'
+}
+
+function buildURL (url, params) {
+  if (!params) {
+    return url
+  }
+
+  const parts = []
+  for (let [key, val] of Object.entries(params)) {
+    if (val === null || typeof val === 'undefined') {
+      return
+    }
+
+    if (!Array.isArray(val)) {
+      val = [val]
+    }
+
+    for (let v of val) {
+      if (isDate(v)) {
+        v = v.toISOString()
+      } else if (isObject(v)) {
+        v = JSON.stringify(v)
```

Comment:
I would prefer if we did not do the JSON.stringify in this case. It's too opinionated.

Reply:
Should we throw on object or ignore it?

Reply:
I think throwing would be a predictable behavior.

Reply:
Done.